### PR TITLE
Make ctor in errorListHelper free threaded

### PR DESCRIPTION
**Pull request by bigfluffycookie**
part of https://github.com/SonarSource/sonarlint-visualstudio/issues/4861

### CODE REVIEW

File: src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs [hunk 1]
```diff
@@ -27,7 +27,6 @@
 using Microsoft.VisualStudio.Shell.TableManager;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Moq;
-using SonarLint.VisualStudio.Core;
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs:30]: Removed as it was unused

File: src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs [hunk 2]
```diff
@@ -38,7 +37,23 @@ public class ErrorListHelperTests
         public void MefCtor_CheckIsExported()
         {
             MefTestHelpers.CheckTypeCanBeImported<ErrorListHelper, IErrorListHelper>(
-                MefTestHelpers.CreateExport<SVsServiceProvider>());
+                MefTestHelpers.CreateExport<IVsUIServiceOperation>());
+        }
+
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs:43]: Added two standard mef tests

File: src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs [hunk 3]
```diff
@@ -60,8 +75,8 @@ public void TryGetRuleIdFromSelectedRow_SingleSonarIssue_ErrorCodeReturned(strin
                 { StandardTableKeyNames.ErrorCode, fullRuleKey }
             });
 
-            var mockErrorList = CreateErrorList(issueHandle);
-            var serviceProvider = CreateServiceProvider(mockErrorList);
+            var errorList = CreateErrorList(issueHandle);
```
Comment (bigfluffycookie) [inline src/Infrastructure.VS.UnitTests/ErrorListHelperTests.cs:78]: I renamed all the "mockErrorList" to errorList as the method doesnt actually return a mock so the naming confused me.

File: src/Infrastructure.VS/ErrorListHelper.cs [hunk 4]
```diff
@@ -33,15 +33,29 @@ namespace SonarLint.VisualStudio.Infrastructure.VS
...
         public bool TryGetRuleIdFromSelectedRow(out SonarCompositeRuleId ruleId)
+        {
+            SonarCompositeRuleId ruleIdOut = null;
+            var result = vSServiceOperation.Execute<SVsErrorList, IErrorList, bool>(
+                errorList =>
+                {
+                    return DoTryGetRuleIdFromSelectedRow(errorList, out ruleIdOut);
+                });
+
+            ruleId = ruleIdOut;
```
[the remark this answers is not in the document]
Comment (duncanp-sonar) [reply src/Infrastructure.VS/ErrorListHelper.cs:53]: That's ok - I've had to do the same in other classes.

**Review by duncanp-sonar — APPROVED** [review]
Nice

**Pull request merged**